src/lib/resources.ts

```diff
@@ -25,6 +26,22 @@ export class Vaults extends OPResource {
         return ObjectSerializer.deserialize(data, "Array<Vault>");
     }
 
+    /**
+     * Searches for all Vaults with exact match on title.
+     *
+     * @param {string} title
+     * @returns {Promise<Vault[]>}
+     * @private
+     */
+     public async getVaultsByTitle(title: string): Promise<Vault[]> {
```

Comment:
Looking at `getItemsByTitle` as a comparison, this function doesn't seem to treat the case in which there are no vaults with the provided title. That also confused me when I tried to get some vaults based on a title, but the token didn't have access to them. The SDK didn't tell me anything about it, so I assumed that the SDK didn't work.

Could you make this function handle this edge case as well? This code snippet inspired from `getItemsByTitle` should do the trick:
```ts
if (!data?.length) {
            return Promise.reject({
                status: 404,
                message: "No Vaults found with title " + title,
            });
        }
```

Reply:
I've implemented exactly in the same way as in go adk
https://github.com/1Password/connect-sdk-go/blob/ac4a2c9c017fcfb76ac97b63b145121926e06fd3/connect/client.go#L203

I see we do such a check but only when we try to get single vault/item. BTW do no see `getItemsByTitle` in js sdk. There is only ability to get single item `getItemByTitle`

Reply:
Please let me know if want to do any changes here.

Reply:
I've dived deeper into the Go SDK "bug" that we thought we've found and actually there's no bug. My code was doing something nasty. The function does throw an error when no items/vaults were found with the provided title:
```
2022/07/25 22:20:31 Found 0 vaults with title "nonexistent-vault-name"
exit status 1
```

So there are no bugs in the Go SDK 😄